src/components/AnimateSharedLayout/index.tsx

```diff
@@ -140,8 +145,30 @@ export class AnimateSharedLayout extends React.Component<SharedLayoutProps> {
         this.stacks.forEach(stack => (stack.snapshot = undefined))
     }
 
+    get isSharedLayoutTreeParent() {
+        return this.tree.order.length > 0
+    }
+
+    forEachChild(cb: (child: HTMLVisualElement) => void) {
+        return this.isSharedLayoutTreeParent
+            ? this.tree.order.forEach(tree => tree.children.forEach(cb))
+            : this.children.forEach(cb)
+    }
+
     updateStacks() {
-        this.stacks.forEach(stack => stack.updateLeadAndFollow())
+        /**
+         * If AnimateSharedLayout is the parent of SharedLayout Trees,
+         * generate new stacks based on the contents of the Lead and Follow trees.
+         */
+        return this.isSharedLayoutTreeParent
+            ? this.updateTree()
+            : this.stacks.forEach(stack => stack.updateLeadAndFollow())
+    }
+
+    updateTree() {
+        this.tree.sortOrder()
```

Comment:
Instead of removing and registering whole trees, just sort the tree in place before we calc the lead and follow so that the latest layoutOrder is last